sdk/evm/evm.go

```diff
@@ -676,14 +676,27 @@ func (client Client) CreateDummyBlocks(
 	}
 	gasPrice := big.NewInt(legacy.BaseFee)
 	txSigner := types.LatestSignerForChainID(chainID)
+	blockNumber, err := client.BlockNumber()
+	if err != nil {
+		return fmt.Errorf("client.BlockNumber failure: %w", err)
+	}
+	nonce, err := client.NonceAt(addr.Hex())
+	if err != nil {
+		return fmt.Errorf("client.NonceAt failure: %w", err)
+	}
 	for i := 0; i < numBlocks; i++ {
-		prevBlockNumber, err := client.BlockNumber()
-		if err != nil {
+		// it may be the case that we hit an outdated node with the rpc, so lets not fully trust the API
+		if blockNumberFromAPI, err := client.BlockNumber(); err != nil {
 			return fmt.Errorf("client.BlockNumber failure at step %d: %w", i, err)
+		} else if blockNumberFromAPI > blockNumber {
+			// changes from outside
+			blockNumber = blockNumberFromAPI
 		}
-		nonce, err := client.NonceAt(addr.Hex())
-		if err != nil {
+		if nonceFromAPI, err := client.NonceAt(addr.Hex()); err != nil {
 			return fmt.Errorf("client.NonceAt failure at step %d: %w", i, err)
+		} else if nonceFromAPI > nonce {
+			// changes from outside
+			nonce = nonceFromAPI
```

Comment:
same here why not we always use nonceFromAPI?

Reply:
same response. nonce from API can be outdated, in this case we trust in local logic. on the other hand,
if the nonce is not outdated but > local one, we trust on the API